commit cadf423848252a6cf936c5f20db599b4ee26ea10
Author: Gawain Lynch <gawain.lynch@gmail.com>
Date:   Fri Aug 15 08:32:13 2014 +0100

    Check for callers URL protocol incorrectly misses that on some systems HTTP systems may return $_SERVER['HTTPS'] == 'off'
    - IIS ISAPI
    - NginX 'fastcgi_param HTTPS off;'

diff --git a/hybridauth/Hybrid/Provider_Adapter.php b/hybridauth/Hybrid/Provider_Adapter.php
index 56d0210..5f0b1a7 100644
--- a/hybridauth/Hybrid/Provider_Adapter.php
+++ b/hybridauth/Hybrid/Provider_Adapter.php
@@ -130,7 +130,7 @@ class Hybrid_Provider_Adapter
 		if (empty(Hybrid_Auth::$config["base_url"])) {
 	        // the base url wasn't provide, so we must use the current
 	        // url (which makes sense actually)
-			$url  = empty($_SERVER['HTTPS']) ? 'http' : 'https';
+			$url  = empty($_SERVER['HTTPS']) || $_SERVER['HTTPS'] == 'off' ? 'http' : 'https';
 			$url .= '://' . $_SERVER['HTTP_HOST'];
 			$url .= $_SERVER['REQUEST_URI'];
 			$HYBRID_AUTH_URL_BASE = $url;
